### ISSUE TITLE
valgrind test of life1d.p4 

### ISSUE THREAD

**Issue by wboeke** [SirWumpus/post4#6]
If you run life1d.p4 from the examples, then valgrind will report 2 issues. I found out (via an other forth) that the problem is'nt the fault of post4, it's from the forth code itself! The last line as drawn by life1d.p4 uses an index with a value of 1 too much. I repaired this in line 28 by an extra test:
```
CHARS dup life1d_size < if life1d_curr + C@ then
```
Maybe there is a better way, but this works.

**Comment by SirWumpus** [comment]
Yesterday when I tested this, the 2 reported errors are related.  This issue I hand was in line 53
```
 : life1d_transition ( -- )
-       life1d_size 1 DO
+       life1d_size 1- 1 DO
                I life1d_curr_state             \ value
```
Which addresses the issue keeping the lookups by `life1d_curr_state` within bounds.

**Comment by SirWumpus** [comment]
```
Rule 104 
   0 .###.##.#.#.#.#..#..
   1 .#.#####.#.#.#......
   2 ..##...##.#.#.......
   3 ..##...###.#........
   4 ..##...#.##.........
   5 ..##....###.........
   6 ..##....#.#.........
   7 ..##.....#..........
   8 ..##................
   9 ..##................
  10 ..##................

==11764== 
==11764== HEAP SUMMARY:
==11764==     in use at exit: 0 bytes in 0 blocks
==11764==   total heap usage: 729 allocs, 729 frees, 166,556 bytes allocated
==11764== 
==11764== All heap blocks were freed -- no leaks are possible
==11764== 
==11764== ERROR SUMMARY: 0 errors from 0 contexts (suppressed: 0 from 0)
roche$ valgrind --version
valgrind-3.22.0
roche$ 
```
With the above change made in 0.8.2 there are no reported issues.

**Comment by SirWumpus** [comment]
With additional comment, closing this as already fixed in v0.8.2.
